src/Uno.Toolkit.UI/Controls/NavigationBar/Renderer.cs

```diff
@@ -139,16 +139,50 @@ internal static class RendererHelper
 	{
 		private static readonly WeakAttachedDictionary<DependencyObject, Type> _renderers = new WeakAttachedDictionary<DependencyObject, Type>();
 
-		public static TRenderer GetRenderer<TElement, TRenderer>(this TElement element, Func<TRenderer>? rendererFactory)
+		public static TRenderer? TryGetRenderer<TElement, TRenderer>(this TElement element)
 			where TElement : DependencyObject
+			where TRenderer : class
 		{
-			return _renderers.GetValue(element, typeof(TRenderer), rendererFactory);
+			TRenderer? renderer = null;
+			if (_renderers.GetValue<TRenderer>(element, typeof(TRenderer)) is { } existingRenderer)
+			{
+				renderer = existingRenderer;
+			}
+
+			return renderer;
+		}
+
+		public static void SetRenderer<TElement, TRenderer>(this TElement element, TRenderer? renderer)
+			where TElement : DependencyObject
+		{
+			_renderers.SetValue(element, typeof(TRenderer), renderer);
+		}
+
+		public static void AddOrUpdateRenderer<TElement, TRenderer>(this TElement element, Func<TElement, TRenderer> onCreate, Action<TElement, TRenderer>? onUpdate = null)
+			where TElement : DependencyObject
+			where TRenderer : class
+		{
+			if (_renderers.GetValue<TRenderer>(element, typeof(TRenderer)) is { } renderer)
+			{
+				onUpdate?.Invoke(element, renderer);
+			}
+			else
+			{
+				element.SetRenderer(onCreate(element));
+			}
 		}
 
-		public static TRenderer ResetRenderer<TElement, TRenderer>(this TElement element, Func<TRenderer>? rendererFactory)
+		public static TRenderer GetOrAddRenderer<TElement, TRenderer>(this TElement element, Func<TElement, TRenderer> rendererFactory)
```

Comment:
it seems like common impl of the factory is: `x => AppBarButtonRenderer(x)`
can we create a specialized version based on TElement? or just ditch the generic+lambda completely if there is no other impl?

Reply:
I know at least for Android we are using a different ctor for the AppBarButtonRenderer. So you'd want some sort of GetOrAddDefaultRenderer extension specifically for AppBarButton/NavigationBar instances? Maybe better naming

Reply:
yeah, more like `GetOrAddDefaultRenderer(this AppBarButton btn)`
if you have various options, it could be `GetOrAddAppBarButtonRendererA/B` with A/B being something more context-relevant
what other options do we have currently?